### ISSUE TITLE
Integer overflow in exponent

### ISSUE THREAD

**Issue by ksvanhorn** [basvandijk/scientific#62]
scientific 1 (maxBound::Int) * scientific 1 (maxBound::Int) evaluates to 1.0e-2.

**Comment by basvandijk** [comment]
Yes, this is unfortunate. `Scientific` uses an `Int` to represent the exponent so it suffers from the same integer overflows that `Int` suffers from. I've an old TODO to switch the exponent to `Integer` to make `Scientific` truly arbitrary precision.

**Comment by basvandijk** [comment]
I started doing some work on this on the [integer-exponent](https://github.com/basvandijk/scientific/tree/integer-exponent) branch.

**Comment by mbj** [comment]
I'd prefer we throw an impure error instead of producing wrong values, till we can get arbitrary precision would a patch be accepted that calls `Prelude(error)` on any operation that would result in corruption?
